### PR TITLE
Fix/refactor page header

### LINKED ISSUE #63: Refactor Page Header Into Fragment
We need this as a fragment since this code is used on multiple pages e.g. main page, create book, update book, etc. This new fragment controller method should also fetch the currently logged in user and display their first and last name as well as an avatar with the first letter of their first name capitalized to replace the template user we are currently displaying in our header bar (currently showing the placeholder user name and avatar for "Valerie Luna").

### PR DESCRIPTION
**Pull request by abungay**
## Issues
Fixes #63 

## Description of changes
- Show the currently logged in user first and last name in the page header along with an avatar that is their first and last name initials.
- If no user is logged in then a Login link is shown instead of the drop-down user menu.
- Refactored the page header from layout.html into a fragment at fragments/_pageHeader.html

## Screenshots (If Applicable)
Screen Shot 2019-04-03 at 3 40 50 AM
Screen Shot 2019-04-03 at 3 42 03 AM

## Checklist
- [ ] I have unit tested my code.
- [ ] I have integration tested my code.
- [x] All new and existing tests passed.
- [ ] I have [updated the diagrams](https://github.com/amazin-team/Amazin-online-bookstore/wiki/Updating-the-diagrams) ModelsUMLClassDiagram and EntityRelationshipDiagram (if necessary).

### CODE REVIEW

**Review by zaidoon1 — DISMISSED** [review]

**Review by zaidoon1 — APPROVED** [review]

**Review by MohamedZalat — APPROVED** [review]

File: src/main/resources/templates/fragments/_pageHeader.html [hunk 1]
```diff
@@ -0,0 +1,104 @@
+<div th:fragment="pageHeader" xmlns:th="http://www.w3.org/1999/xhtml">
```
Comment (LauraMachado21) [inline src/main/resources/templates/fragments/_pageHeader.html:1]: Nice refactoring!

File: src/main/java/amazin/controller/FragmentController.java [hunk 2]
```diff
@@ -1,21 +0,0 @@
-package amazin.controller;
```
Comment (LauraMachado21) [inline src/main/java/amazin/controller/FragmentController.java:1]: Good catch!

File: src/main/java/amazin/service/SecurityServiceImpl.java [hunk 3]
```diff
@@ -22,9 +23,11 @@
 
     @Override
     public String findLoggedInEmail() {
-        Object userDetails = SecurityContextHolder.getContext().getAuthentication().getDetails();
-        if (userDetails instanceof UserDetails) {
-            return ((UserDetails)userDetails).getUsername();
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication.getPrincipal() instanceof UserDetails) {
```
Comment (LauraMachado21) [inline src/main/java/amazin/service/SecurityServiceImpl.java:27]: Can you please explain what was wrong with the initial implementation?

Comment (abungay) [reply src/main/java/amazin/service/SecurityServiceImpl.java:27]: The initial implementation, even when the user was logged in, for some reason was always returning null. This modification properly returns back the email if the user is logged in or null as the email if no user is logged in.

**Pull request merged**